### PR TITLE
Issue/133 handle relations params

### PR DESCRIPTION
**Pull request by vallo87**
implemented new editing feature for relation params
- new editing view in panel
- listing all params in realtion-view (or type of param if not set)

### CODE REVIEW

File: src/Controller/ModulesController.php [hunk 1]
```diff
@@ -203,6 +203,7 @@ public function save() : ?Response
                 foreach ($requestData['api'] as $api) {
                     extract($api); // method, id, type, relation, relatedIds
                     if (in_array($method, ['addRelated', 'removeRelated', 'replaceRelated'])) {
+                        // dd($requestData);
```
Comment (stefanorosanelli) [inline src/Controller/ModulesController.php]: remove debug code

File: src/Controller/ModulesController.php [hunk 2]
```diff
@@ -321,6 +322,34 @@ protected function updateMediaUrls(array &$response) : void
...
+     * Json response
+     *
+     * @param string|int $id the object identifier.
+     * @param string $relation the relating name.
+     * @return void
+     */
+    public function relationData($id, string $relation) : void
+    {
+        $this->request->allowMethod(['get']);
+        $response = null;
```
Comment (stefanorosanelli) [inline src/Controller/ModulesController.php]: Unnecessary assignement, line can be dropped.

File: src/Controller/ModulesController.php [hunk 3]
```diff
@@ -321,6 +322,34 @@ protected function updateMediaUrls(array &$response) : void
...
+            $this->log($error, LogLevel::ERROR);
+
+            $this->set(compact('error'));
+            $this->set('_serialize', ['error']);
+
+            return;
+        }
+
+        $this->set((array)$response);
+        $this->set('_serialize', array_keys($response));
```
Comment (stefanorosanelli) [inline src/Controller/ModulesController.php]: Hint: use `$this->set('_serialize', true);` to serialize everything

File: src/Template/Layout/js/app/directives/datepicker.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ import 'flatpickr/dist/flatpickr.min.css';
 
 const datepickerOptions = {
     enableTime: false,
-    dateFormat: "Y-m-d H:i",
+    dateFormat: "Z",
```
Comment (stefanorosanelli) [inline src/Template/Layout/js/app/directives/datepicker.js]: why change format to UTC?
this also has a bad side effect: validation error on BE4 API due to a bug

**Comment by vallo87** [comment]
If relation has no params the button edit params will not show up

**Pull request merged**